### PR TITLE
CLI: extract bin flag def from index

### LINKED ISSUE #2092: Make the ChromeLauncher API into a real thing.
<details>
<summary>
Here's a few incarnations I'm seeing of people using the ChromeLauncher API:
</summary>

image
image
image
image
image
image
image

and this dude who's just like "wtf is autoselect?": https://github.com/siddharthkp/lighthouse-ci/blob/c3938a2523434f26ea7963b0546fb241d4f21d36/index.js

</details>

_(click above)_

and not to mention the example in lighthouse's readme [which is like whaaaaa.](https://github.com/GoogleChrome/lighthouse#using-programmatically).

There's a lot of variety and nothing I'm seeing in those snippets actually _looks correct._

**Feels like we could use chrome-launcher module that has a real first-class API.**

Something with a very clear API surface to handle launch, getting the port number (if set with 0), ability to kill.. 

It'd also be nice to have sigint [kill the process automatically as well](https://github.com/paulirish/pwmetrics/blob/4070b62ccc0e84c25fd81ce6e3e32c284f8e639b/lib/index.ts#L61-L71). (our poor manual-chrome-launcher doesn't have this currently).

wdyt?

+ @samccone @ebidel

### PR DESCRIPTION
**Pull request by samccone**
* Extract Flags interface instead of defining inline in the function defs.
* Extract bin flags into a standalone file since this is more about config and less about logic.

ref #2092

depends on 
https://github.com/GoogleChrome/lighthouse/pull/2173

### CODE REVIEW

File: lighthouse-cli/bin.ts [hunk 1]
```diff
@@ -27,107 +27,37 @@ import {ChromeLauncher} from './chrome-launcher';
...
-        .default('output', Printer.GetValidOutputOptions()[Printer.OutputMode.html])
-        .default('port', 9222)
-        .default('max-wait-for-load', Driver.MAX_WAIT_FOR_FULLY_LOADED)
-        .check((argv: {listAllAudits?: boolean, listTraceCategories?: boolean, _: Array<any>}) => {
-          // Make sure lighthouse has been passed a url, or at least one of --list-all-audits
-          // or --list-trace-categories. If not, stop the program and ask for a url
-          if (!argv.listAllAudits && !argv.listTraceCategories && argv._.length === 0) {
-            throw new Error('Please provide a url');
-          }
+interface Flags {
```
Comment (patrickhulce) [inline lighthouse-cli/bin.ts]: shouldn't this interface be defined and exported in `cli-flags.ts`?

Comment (paulirish) [inline lighthouse-cli/bin.ts]: separately, thanks for extracting this. the # of times this type was inlined was killing me.

**Comment by samccone** [comment]
All updated PTAL.

File: lighthouse-cli/bin.ts [hunk 2]
```diff
@@ -322,39 +216,35 @@ function saveResults(results: Results,
...
+      await chromeLauncher!.kill();
     }
 
-    // This will be reenabled once flags are extracted to standalone interface.
-    // clang-format off
+    return handleError(err);
+  }
+}
+
+// This will be reenabled once flags are extracted to standalone interface.
```
Comment (paulirish) [inline lighthouse-cli/bin.ts]: is that now?

File: lighthouse-cli/cli-flags.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
+const yargs = require('yargs');
```
Comment (paulirish) [inline lighthouse-cli/cli-flags.ts]: add license header

use strict?

File: lighthouse-cli/bin.ts [hunk 4]
```diff
@@ -186,29 +84,27 @@ if (cliFlags.output === Printer.OutputMode[Printer.OutputMode.json] && !cliFlags
...
-  return randomPort.getRandomPort().then(portNumber => {
-    flags.port = portNumber;
-    log.verbose('Lighthouse CLI', `Using generated port ${flags.port}.`);
+    log.verbose('Lighthouse CLI', 'Generating random port.');
+    return randomPort.getRandomPort().then(portNumber => {
+      flags.port = portNumber;
+      log.verbose('Lighthouse CLI', `Using generated port ${flags.port}.`);
+    });
   });
-  })
```
Comment (paulirish) [inline lighthouse-cli/bin.ts]: welcome back, semicolon.

File: lighthouse-cli/cli-flags.ts (hunk 3 above)
Comment (samccone) [inline lighthouse-cli/cli-flags.ts]: done

File: lighthouse-cli/bin.ts (hunk 2 above)
Comment (samccone) [inline lighthouse-cli/bin.ts]: done

File: lighthouse-cli/bin.ts (hunk 4 above)
Comment (samccone) [inline lighthouse-cli/bin.ts]: 👋

**Pull request merged**